### PR TITLE
Draw block quotes inside lists

### LINKED ISSUE #449: Add mixed lists and blockquote support.
We should support mixing lists and blockquotes.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #449 

This PR adds support for drawing block quotes inside lists.

To test:
 - Run the demo app with sample content
 - Scroll to the place where lists are mixed with block quotes and see if the presentation is done correctly
 - Open the empty demo app
 - write something
 - Tap on list formatting
 - Tap on block quote formatting
 - Check if presentation is correct.

### CODE REVIEW

**Comment by SergioEstevao** [comment]
@jleandroperez ready for review.

**Comment by jleandroperez** [comment]
Couple notes!

A. Hitting *Tab* causes the bullet to indent, but the blockquote is not moving
B. We should probably cut the "nested" style after hitting enter. IE:
  1. Add a List
  2. Toggle Blockquote
  3. Enter a line of text
  4. Hit Enter
  In this scenario we should probably remove the blockquote from the next bullet (We're currently carrying it over).

I'll keep on testing, hopefully nothing else will show up!

**Comment by jleandroperez** [comment]
Another note!: We should probably add a little bit of padding between the Bullet and the Blockquote, since they look super close,

**Comment by SergioEstevao** [comment]
> Another note!: We should probably add a little bit of padding between the Bullet and the Blockquote, since they look super close,

Vertically or horizontally?

**Comment by SergioEstevao** [comment]
> A. Hitting Tab causes the bullet to indent, but the blockquote is not moving

Is this scenario where block quote comes first or lists come first?

**Comment by jleandroperez** [comment]
@SergioEstevao Compiling all of the comments, and adding extra details on each one of them.

We can always address the rest in different issues, absolutely cool with it!  But we should probably address, at least, Comment C here, since it's related to the drawing snippets.

Thanks sir, nice job!!!

### Comment A: Nested Lists
Hitting Tab causes the bullet to indent, but the blockquote is not moving. Steps:

1. Toggle Ordered List
2. Toggle Blockquote
3. Hit Tab

As a result the Ordered List will get shifted rightwards, but the blockquote will remain in place.

### Comment B: Newline
We should probably cut the "nested" style after hitting enter. IE:

1. Add a List
2. Toggle Blockquote
3. Enter a line of text
4. Hit Enter

In this scenario we should probably remove the blockquote from the next bullet (We're currently carrying it over).

### Comment C: Spacing
Adding an image below, comparing Aztec and Calypso. Spacing should proooobably be horizontal (between the bullet and the blockquote).

image

**Comment by SergioEstevao** [comment]
@lantean I agree with except for C all the others should be separated bugs.

A) is the result of list being added to the end of properties when indenting instead of following the last list

B) It's a general logic bug/issue we need to find how the other editor behave on these scenarios.

**Comment by SergioEstevao** [comment]
@jleandroperez I improved the layout, give it another look please.

File: Aztec/Classes/TextKit/ParagraphStyle.swift [hunk 1]
```diff
@@ -183,6 +183,39 @@ open class ParagraphStyle: NSMutableParagraphStyle, CustomReflectable {
...
+
+        return CGFloat(depth) * Metrics.listTextIndentation
+    }
+
+    /// The amount of indent for the list of the paragraph if any.
+    ///
+    public var listIndent: CGFloat {
+        let listAndBlockquotes = self.properties.filter({ (property) -> Bool in
+            return property is Blockquote || property is TextList
+        })
```
Comment (jleandroperez) [inline Aztec/Classes/TextKit/ParagraphStyle.swift]: Few notes!

- No need for `self`
- No need to wrap up the last closure within braces
- No need to define the closure's return type

```
let listAndBlockquotes = properties.filter { property in
    return property is Blockquote || property is TextList
}
```

File: Aztec/Classes/TextKit/ParagraphStyle.swift [hunk 2]
```diff
@@ -183,6 +183,39 @@ open class ParagraphStyle: NSMutableParagraphStyle, CustomReflectable {
         }
     }
 
+    /// The amount of indent for the blockquote of the paragraph if any.
+    ///
+    public var blockquoteIndent: CGFloat {
+        let blockquoteIndex = self.properties.filter({ (property) -> Bool in
```
Comment (jleandroperez) [inline Aztec/Classes/TextKit/ParagraphStyle.swift]: - No need for `self`
- No need to define the return type

```
let blockquoteIndex = properties.filter({ property in
```

**Comment by jleandroperez** [comment]
@SergioEstevao the new indentation feels a bit too much. Can we shrink that by 25% (or so) perhaps?

simulator screen shot jul 20 2017 10 10 55 am

Awesome changes BTW!!

**Comment by SergioEstevao** [comment]
@jleandroperez addressed your comments, can you have another look.

**Pull request merged**